tests/test_jax.py

```diff
@@ -241,24 +241,139 @@ def internal_gather(operand, start_indices):
     run_and_compare_specific_input(wrapped_gather(dimension_numbers, (1, 5)), (operand, start_indices))
 
 
+def test_complex_gather():
+    from jax.lax import GatherDimensionNumbers
+
+    def wrapped_gather(dimension_numbers, slice_sizes):
+        @jax.jit
+        def internal_gather(operand, start_indices):
+            return jax.lax.gather(
+                operand=operand,
+                start_indices=start_indices,
+                dimension_numbers=dimension_numbers,
+                slice_sizes=slice_sizes,
+            )
+        return internal_gather
+
+    start_indices = [
+                     [
+                      [[0, 0], [1, 0], [2, 1]],
+                      [[0, 1], [1, 1], [0, 9]]
+                     ],
+                     [
+                      [[0, 0], [2, 1], [2, 2]],
+                      [[1, 2], [0, 1], [1, 0]]
+                     ]
+                    ]
+    start_indices = jnp.array(start_indices, dtype=jnp.int32)
+    operand = jnp.arange(1, 49).reshape((2, 3, 4, 2))
+    dimension_numbers = GatherDimensionNumbers(
+        offset_dims = (3, 4),
+        collapsed_slice_dims = (1,),
+        operand_batching_dims = (0,),
+        start_indices_batching_dims = (0,),
+        start_index_map = (2, 1),
+    )
+    run_and_compare_specific_input(wrapped_gather(dimension_numbers, (1, 1, 1, 2)), (operand, start_indices))
+
+    start_indices = jnp.concatenate((start_indices, start_indices[::-1, 1:]), 1)
+    dimension_numbers = GatherDimensionNumbers(
+        offset_dims = (3, 4),
+        collapsed_slice_dims = (),
+        operand_batching_dims = (0, 1),
+        start_indices_batching_dims = (0, 1),
+        start_index_map = (3, 2),
+    )
+    result = wrapped_gather(dimension_numbers, (1, 1, 1, 1))(operand, start_indices)
+    run_and_compare_specific_input(wrapped_gather(dimension_numbers, (1, 1, 1, 1)), (operand, start_indices))
+
+
+def test_simple_scatter():
```

Comment:
This is good, it would be nice to also have another test-function that test more complicated forms of scatter using the `arr.at[...].set()` functionality, to test it works for more dimensions.